src/components/Files/Files.js

```diff
@@ -43,63 +48,156 @@ const Files = ({
+        }
+      }))
+
+      const result = await fetchFile(item).catch(error => {
+        setPageData(state => ({
+          ...state,
+          selectedRowData: {
+            ...state.selectedRowData,
+            [item.db_key]: {
+              ...state.selectedRowData[item.db_key],
+              error,
+              loading: false
+            }
+          }
+        }))
+      })
+
+      if (result?.length > 0) {
+        setYamlContent(state => ({
+          ...state,
+          selectedRowData: result
+        }))
+        setPageData(state => {
+          return {
+            ...state,
+            selectedRowData: {
+              ...state.selectedRowData,
+              [item.db_key]: {
+                content: [...generateArtifacts(filterArtifacts(result))],
+                error: null,
+                loading: false
+              }
+            }
+          }
+        })
+      }
+    },
+    [fetchFile]
+  )
+
+  const handleExpandRow = useCallback((item, isCollapse) => {
+    if (isCollapse) {
+      setYamlContent(state => ({
+        ...state,
+        selectedRowData: []
+      }))
+    }
+  }, [])
+
   useEffect(() => {
-    fetchData({ project: match.params.projectName })
+    fetchData({ project: match.params.projectName, onEntering: true })
 
     return () => {
+      setGroupFilter('')
       setFiles([])
       removeFiles()
       setSelectedFile({})
+      setYamlContent({
+        allData: [],
+        selectedRowData: []
+      })
     }
   }, [fetchData, match.params.projectName, removeFiles])
 
   useEffect(() => {
-    if (match.params.name && artifactsStore.files.length !== 0) {
-      const { name } = match.params
-
-      const [searchItem] = artifactsStore.files.filter(
-        item => item.key === name
-      )
+    if (artifactsStore.filter.tag === 'latest') {
+      setGroupFilter('name')
+    } else if (groupFilter.length > 0) {
+      setGroupFilter('')
+    }
+  }, [match.params.pageTab, groupFilter.length, artifactsStore.filter])
 
-      if (!searchItem) {
-        history.push(`/projects/${match.params.projectName}/files`)
-      } else {
-        const file = maxBy(
-          searchItem.data.filter(item => {
-            if (searchItem.link_iteration) {
-              const { link_iteration } = searchItem.link_iteration
+  useEffect(() => {
+    setPageData(state => ({
+      ...state,
+      handleRequestOnExpand,
+      handleRemoveRequestData: handleRemoveFile
+    }))
+  }, [handleRemoveFile, handleRequestOnExpand])
 
-              return link_iteration === item.iter
-            }
+  useEffect(() => {
+    if (match.params.name) {
+      const { name } = match.params
+      let artifacts = []
 
-            return true
-          }),
-          'updated'
-        )
+      if (artifactsStore.files.selectedRowData.content[name]) {
+        artifacts = artifactsStore.files.selectedRowData.content[name]
+      } else {
+        artifacts = artifactsStore.files.allData
+      }
```

Comment:
Equivalent to:
```suggestion
      artifacts = artifactsStore.files.selectedRowData.content[name] || artifactsStore.files.allData
```
Or, if you only care for `null` and `undefined` (and not for `0`, `false`, `''`, `NaN`), then:
```suggestion
      artifacts = artifactsStore.files.selectedRowData.content[name] ?? artifactsStore.files.allData
```